storage/adls_gen1_mount.go

```diff
@@ -30,7 +31,7 @@ func (m AzureADLSGen1Mount) Config() map[string]string {
 
 		m.PrefixType + ".oauth2.client.id":   m.ClientID,
 		m.PrefixType + ".oauth2.credential":  fmt.Sprintf("{secrets/%s/%s}", m.SecretScope, m.SecretKey),
-		m.PrefixType + ".oauth2.refresh.url": fmt.Sprintf("https://login.microsoftonline.com/%s/oauth2/token", m.TenantID),
+		m.PrefixType + ".oauth2.refresh.url": fmt.Sprintf("%s/%s/oauth2/token", common.AzureEnvironmentFromARMEnvironment().ActiveDirectoryEndpoint, m.TenantID),
```

Comment:
Don't just rely on environment, because we can specify azure environment in provider block through "azure_env" property (or something like that one) and this code will respect only the ARM_ENVIRONMENT. 

This will require adding DatabricksClient argument to Config method, so you can retrieve correctly configured c.AzureAuth. getAzureEnvironment. And modifying the mount interface.

Reply:
updated, unit tests are missing I will look into that as well.